dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/TCPUtils.java

```diff
@@ -81,28 +161,32 @@ private static ChannelFuture bind(NettyFrameworkImpl framework, ServerBootstrapE
                             Tr.debug(tc, "sleep caught InterruptedException.  will proceed.");
                         }
                     }
-                    bind(framework, bootstrap, inetHost, inetPort, bindListener, retryCount - 1);
+                    open(framework, bootstrap, config, newHost, inetPort, openListener, retryCount - 1);
                 } else {
-                    Tr.error(tc, TCPMessageConstants.BIND_ERROR,
-                            new Object[] { channelName, inetHost, String.valueOf(inetPort), 
-                                    bindFuture.cause().getMessage() });
+                    if (config.isInbound()) {
+                        Tr.error(tc, TCPMessageConstants.BIND_ERROR, new Object[] { config.getExternalName(), newHost,
+                                String.valueOf(inetPort), openFuture.cause().getMessage() });
+                    } else {
+                        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                            Tr.debug(tc, TCPMessageConstants.BIND_ERROR, new Object[] { config.getExternalName(),
+                                    newHost, String.valueOf(inetPort), openFuture.cause().getMessage() });
+                        }
+                    }
                 }
             }
         });
-        return bindFuture;
+        return openFuture;
     }
 
-    public static ChannelFuture start(NettyFrameworkImpl framework, ServerBootstrapExtended bootstrap, String inetHost, int inetPort, ChannelFutureListener bindListener) throws NettyException {
-        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-            Tr.debug(tc, "start (TCP): attempt to bind a channel at host " + inetHost + " port " + inetPort);
-        }
+    private static ChannelFuture startHelper(NettyFrameworkImpl framework, AbstractBootstrap bootstrap,
+            TCPConfigurationImpl config, String inetHost, int inetPort, ChannelFutureListener openListener)
+            throws NettyException {
         try {
             framework.runWhenServerStarted(new Callable<ChannelFuture>() {
                 @Override
                 public ChannelFuture call() {
-                    TCPConfigurationImpl config = (TCPConfigurationImpl) bootstrap.getConfiguration();
-                    int bindRetryCount = config.getPortOpenRetries();
-                    return bind(framework, bootstrap, inetHost, inetPort, bindListener, bindRetryCount);
+                    return open(framework, bootstrap, config, inetHost, inetPort, openListener,
+                            config.getPortOpenRetries());
```

Comment:
As we are running this asynchronously, it is a bit brave to always assume the config instance is still set - what about just resolving the number of retries now?